src/os/bluestore/BlueStore.cc

```diff
@@ -8827,6 +8827,131 @@ bool BlueStore::has_null_manager() const
   return (fm && fm->is_null_manager());
 }
 
+int BlueStore::mount_readonly()
+{
+  int r = _mount_readonly();
+  if (r < 0) {
+    return r;
+  }
+  r = _open_collections();
+  if (r < 0) {
+    return r;
+  }
+  auto shutdown_cache = make_scope_guard([&] {
+    if (!mounted) {
+      _shutdown_cache();
+    }
+  });
+
+  _kv_start();
+  auto stop_kv = make_scope_guard([&] {
+    if (!mounted) {
+      _kv_stop();
+    }
+  });
+
+  r = _deferred_replay();
+  if (r < 0) {
+    return r;
+  }
+  mempool_thread.init();
+  mounted = true;
+  return r;
+}
+
+int BlueStore::_mount_readonly()
```

Comment:
I feel a need of some comment of how this read-only is achieved.
I guess it is some reliance on the fact that if allocator is not initialized then nothing new can be written?

Reply:
Not really - major actions for opening in RO mode are the follows:
- _open_db calls RocksDB::open_read_only if read_only param provided. Which in turn results in calling 
status = rocksdb::DB::OpenForReadOnly(opt, path, &db);
- omitting calling submit_transaction_sync in kv_sync_thread if opened in RO
- omitting sync_metadata call in BlueFS::umount if opened in RO mode.
- omitting allocator creation which is rather an auxiliary mean to ensure no writes to new target locations occur. Writing to already allocated extents, e.g. ones bound to deferred ops are allowed though.